feat: add warning page if users haven't verified their e-mail

iwalmsley opened the pull request.

iwalmsley commented:
Working on why this fails sometimes

andrew-longmore-tessella commented on code/workspaces/web-app/src/components/welcome/VerifyEmail.js:
```diff
@@ -0,0 +1,53 @@
+const VerifyEmail = ({ classes }) => (
+  <div className={classes.bar}>
+    <img className={classes.logo} src={datalabsLogo} alt="DataLabs-Logo" />
+    <Typography className={classes.tagLine} variant="h6">{tagLine}</Typography>
```
This renders as all capitals, which I found a bit 'shouty'?

andrew-longmore-tessella commented on code/workspaces/web-app/src/components/welcome/VerifyEmail.js:
```diff
@@ -0,0 +1,53 @@
+const VerifyEmail = ({ classes }) => (
+  <div className={classes.bar}>
+    <img className={classes.logo} src={datalabsLogo} alt="DataLabs-Logo" />
+    <Typography className={classes.tagLine} variant="h6">{tagLine}</Typography>
```
This renders as all capitals, which I found a bit 'shouty'?

alex-lee-tessella replied:
I pointed Iain to the cover from the public app to base it on which is where the text style comes from. I agree that with the extra text it becomes a bit "shouty". I had a little play with the styling in some spare time I had the other day and altered it to the attached image. I have it as code changes so if people like it or want to take it forward as a base then I can merge it into this branch.

![image](https://user-images.githubusercontent.com/52826110/95864548-6dc9e680-0d5d-11eb-920b-b1732ad3a493.png)

iwalmsley replied:
Yes that would be great if you don't mind

iwalmsley commented:
I have updated this, changed the verified link to re-log in as a cookie isn't issued until a user logs in successfully. Have tried to test on different browsers etc which seems to be working okay.  Open to a different approach around how the request is routed/served though, can't think of a better way unfortunately.

alex-lee-tessella commented on code/workspaces/web-app/src/components/welcome/VerifyEmail.spec.js:
```diff
@@ -0,0 +1,21 @@
+
+jest.mock('../../auth/auth');
+getAuth.mockImplementation(() => ({
+  login: jest.fn(),
```
I find it useful to name mock functions as the name appears in any snapshots produced. Makes it easier to know you are passing the right function to the right prop of any child components!

e.g. `login: jest.fn().mockName('login')`

alex-lee-tessella reviewed (approved):
Looks good to me. Might be worth adding in the mock name, but not too bothered by it as there aren't a lot of other mocks going into the component!
